### PR TITLE
DPC-1312 job status

### PR DESCRIPTION
**Pull request by MrBilnon**
### Fixes [DPC-1312](https://jira.cms.gov/browse/DPC-1312)

This PR mimics the java code to get the status of the job.

### Proposed Changes

* add two methods, one to get the batch and one to get the files for the batch
* add a job client so can point to shared job service in the future
* process jobs and files like the java code to derive the status

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### CODE REVIEW

File: dpc-go/dpc-api/src/v2/job.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	for _, f := range files {
+		resourceType := f.ResourceType
+
+		if errorList && resourceType == "OperationOutcome" {
+			output := model.Output{
+				Type: f.ResourceType,
+				URL:  fmt.Sprintf("%s/Data/%s.ndjson", "", f.FormOutputFileName()),
+			}
+			outputs = append(outputs, output)
+		} else if !errorList && resourceType != "OperationOutcome" {
```
Comment (dhgreene) [inline dpc-go/dpc-api/src/v2/job.go:114]: So the logic here is to ignore anything where the `resourceType` and `errorList` vars don't match?  Should we log an error in that unexpected case?

File: dpc-go/dpc-attribution/src/service/v1/job.go [hunk 2]
```diff
@@ -53,91 +51,153 @@ func NewJobService(pr repository.PatientRepo, jr repository.JobRepo) service.Job
...
+	////for _, s := range batches {
+	////	batchIDs = append(batchIDs, s.BatchID)
+	////}
+	////
+	////files, err := js.jr.FindBatchFilesByBatchID(r.Context(), batchIDs)
+	////
+	////if err != nil {
+	////	log.Error("Failed to find batch files", zap.Error(err))
+	////	boom.Internal(w, r)
+	////}
```
Comment (dhgreene) [inline dpc-go/dpc-attribution/src/service/v1/job.go:200]: Ready to remove these?

File: dpc-go/dpc-attribution/src/repository/job_repository_v1_test.go [hunk 3]
```diff
@@ -69,53 +69,35 @@ func (suite *JobRepositoryV1TestSuite) TestInsert() {
...
+	).WillReturnResult(sqlmock.NewResult(1, 1))
 	mock.ExpectCommit()
-	job, err := repo.Insert(ctx, batches)
-	if err := mock.ExpectationsWereMet(); err != nil {
-		suite.T().Errorf("there were unfulfilled expectations: %s", err)
-	}
+	job, err := repo.Insert(ctx, "12345", batches)
+	//if err := mock.ExpectationsWereMet(); err != nil {
+	//	suite.T().Errorf("there were unfulfilled expectations: %s", err)
+	//}
```
Comment (dhgreene) [inline dpc-go/dpc-attribution/src/repository/job_repository_v1_test.go:98]: More to delete?

File: dpc-go/dpc-api/src/client/job_client.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"github.com/CMSgov/dpc/api/logger"
+	middleware2 "github.com/CMSgov/dpc/api/middleware"
```
[the remark this answers is not in the document]
Comment (MrBilnon) [reply dpc-go/dpc-api/src/client/job_client.go:7]: no doubt, I just have no idea what to call this package of stuff

File: dpc-go/dpc-api/src/client/job_client.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+// JobClientImpl is a struct to hold the retryablehttp client and configs
+type JobClientImpl struct {
+	config     JobConfig
+	httpClient *retryablehttp.Client
+}
+
+// NewJobClient initializes the retryable client and returns a reference to the attribution client
+func NewJobClient(config JobConfig) JobClient {
+	client := retryablehttp.NewClient()
+	client.RetryMax = config.Retries
```
[the remark this answers is not in the document]
Comment (MrBilnon) [reply dpc-go/dpc-api/src/client/job_client.go:37]: it appears to be shorter, I've tested this by not having a running attribution service and it takes about 20 seconds to do all 3 retries

**Review by dhgreene — APPROVED** [review]
🕺

**Pull request merged**